fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -0,0 +1,249 @@
+        return super.visitGroupPlan(groupPlan, context);
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalAggregate(PhysicalAggregate<Plan> agg, Void context) {
+        return computeAggregate(agg);
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalOlapScan(PhysicalOlapScan olapScan, Void context) {
+        return computeScan(olapScan);
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalHeapSort(PhysicalHeapSort<Plan> sort, Void context) {
+        return groupExpression.getCopyOfChildStats(0);
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> hashJoin, Void context) {
+        return HashJoinEstimation.estimate(groupExpression.getCopyOfChildStats(0),
+                groupExpression.getCopyOfChildStats(1),
+                hashJoin.getCondition().get(), hashJoin.getJoinType());
+    }
+
+    // TODO: We should subtract those pruned column, and consider the expression transformations in the node.
+    @Override
+    public StatsDeriveResult visitPhysicalProject(PhysicalProject<Plan> project, Void context) {
+        return computeProject(project);
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalFilter(PhysicalFilter<Plan> filter, Void context) {
+        return computeFilter(filter);
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalDistribution(PhysicalDistribution<Plan> distribution,
+            Void context) {
+        return groupExpression.getCopyOfChildStats(0);
+    }
+
+    private StatsDeriveResult computeFilter(Filter filter) {
+        StatsDeriveResult stats = groupExpression.getCopyOfChildStats(0);
+        FilterSelectivityCalculator selectivityCalculator =
+                new FilterSelectivityCalculator(stats.getSlotRefToColumnStatsMap());
+        double selectivity = selectivityCalculator.estimate(filter.getPredicates());
+        stats.multiplyDouble(selectivity);
+        return stats;
+    }
+
+    // TODO: 1. Subtract the pruned partition
+    //       2. Consider the influence of runtime filter
+    //       3. Get NDV and column data size from StatisticManger, StatisticManager doesn't support it now.
+    private StatsDeriveResult computeScan(Scan scan) {
+        Table table = scan.getTable();
+        TableStats tableStats = Utils.execWithReturnVal(() ->
+                Catalog.getCurrentCatalog().getStatisticsManager().getStatistics().getTableStats(table.getId())
+        );
+        Map<Slot, ColumnStats> slotToColumnStats = new HashMap<>();
+        Set<SlotReference> slotSet = scan.getOutput().stream().filter(SlotReference.class::isInstance)
+                .map(s -> (SlotReference) s).collect(Collectors.toSet());
+        for (SlotReference slotReference : slotSet) {
+            String colName = slotReference.getName();
+            if (colName == null) {
+                // TODO: should we throw an exception here?
```

Comment:
I think an error needs to be reported, because the scan node will generate the column name.